src/compas/datastructures/tree/tree.py

```diff
@@ -47,16 +43,16 @@ class TreeNode(Data):
         "type": "object",
         "$recursiveAnchor": True,
         "properties": {
-            "name": {"type": "string"},
             "attributes": {"type": "object"},
             "children": {"type": "array", "items": {"$recursiveRef": "#"}},
         },
-        "required": ["name", "attributes", "children"],
+        "required": ["attributes", "children"],
     }
 
     def __init__(self, name=None, attributes=None):
         super(TreeNode, self).__init__(name=name)
-        self.attributes = attributes or {}
+        if attributes:
```

Comment:
should it not be
```python
def __init__(self, **kwargs):
    super(TreeNode, self).__init__(**kwargs)
    self._parent = None
    ...
```

Reply:
and the same for the tree itself?
and should we not also remove the attributes from the data dict?

Reply:
> should it not be
> 
> ```python
> def __init__(self, **kwargs):
>     super(TreeNode, self).__init__(**kwargs)
>     self._parent = None
>     ...
> ```

right, indeed! updated now

Reply:
> and the same for the tree itself? and should we not also remove the attributes from the data dict?

You mean like this?
```python
@property
    def data(self):
        return {
            # "attributes": self.attributes,
            "children": [child.data for child in self.children],
        }
```

We would lost track of name then, unless we put it back explicitly.
At least for TreeNode isn't it nice to keep the whole attributes dict so there is a bit more flexibility?

Reply:
for the tree node there are two options
1. keep the attributes dict in the data dict and convert the node explicitly to/from data
2. don't include the attributes dict in the data dict and let the encoder/decoder handle to/from (then the attributes are included implicitly)

the file size of 1. is much smaller and is similar to what happens to nodes in a graph and vertices in a mesh. however, there the behaviour is not so explicit since the nodes and vertices are not separate objects.

if a tree node is not meant to exist on its own (like graph nodes and mesh vertices), perhaps we should make sure that they cannot be converted to/from data independently?

Reply:
I'm more in favor of option.1. imo, the node can be create independently but there is no need to serialise them outside context of tree. They also don't need guid, because their structural location in the tree makes them naturally unique. Maybe it should even not inherant from `Data` at all... Well this is distracting the purpose of this PR, shall we keep it for now and open another PR for it?

Reply:
mmm....  actually... option 2 is also not bad........ well let's have this discussion in another PR...

Reply:
currently everything works, so we can merge this and continue the conversation later. but we need to make sure the behaviour is consistent everywhere and 100% transparent.

@gonzalocasas @chenkasirer opinions?

Reply:
> mmm.... actually... option 2 is also not bad........ well let's have this discussion in another PR...

i also prefer option 1 btw

Reply:
> currently everything works, so we can merge this and continue the conversation later. but we need to make sure the behaviour is consistent everywhere and 100% transparent.
> 
> @gonzalocasas @chenkasirer opinions?

yes agree